File: app/src/main/java/com/zwstudio/lolly/android/SearchActivity.java
package com.zwstudio.lolly.android;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.SearchView;
import android.widget.Toast;

import com.zwstudio.lolly.data.RepoDictionary;
import com.zwstudio.lolly.domain.Dictionary;

import roboguice.inject.ContentView;
import roboguice.inject.InjectView;

@ContentView(R.layout.activity_search)
public class SearchActivity extends DrawerActivity {

    @InjectView(R.id.svWord)
    SearchView svWord;
    @InjectView(R.id.wvDictOnline)
    WebView wvDictOnline;
    @InjectView(R.id.wvDictOffline)
    WebView wvDictOffline;

    String word = "";
    boolean webViewFinished = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        configWebView(wvDictOnline);
        configWebView(wvDictOffline);

        wvDictOnline.setVisibility(View.INVISIBLE);
        wvDictOffline.setVisibility(View.INVISIBLE);
        svWord.setQuery(getSettingsViewModel().word, false);
        svWord.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                searchDict(svWord);
                return true;
            }
            @Override
            public boolean onQueryTextChange(String newText) {
                getSettingsViewModel().word = newText;
                return false;
            }
        });
    }

    private void configWebView(WebView wv) {
        wv.getSettings().setJavaScriptEnabled(true); // enable javascript
        Activity activity = this;
        wv.setWebViewClient(new WebViewClient() {
            @Override
            public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
                Toast.makeText(activity, description, Toast.LENGTH_SHORT).show();
            }
        });
    }

    public void searchDict(View view) {
        word = svWord.getQuery().toString();
        wvDictOnline.setVisibility(View.VISIBLE);
        wvDictOffline.setVisibility(View.INVISIBLE);
        Dictionary m = getSettingsViewModel().getSelectedDict();
        String url = RepoDictionary.urlString(m.url, word);
        svWord.post(new Runnable() {
            @Override
            public void run() {
                svWord.clearFocus();
            }
        });
        wvDictOnline.loadUrl(url);
    }

}

File: app/src/main/java/com/zwstudio/lolly/android/SearchActivity.kt
package com.zwstudio.lolly.android

import android.view.View
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.SearchView
import android.widget.Toast
import com.zwstudio.lolly.data.SearchViewModel
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.Bean
import org.androidannotations.annotations.EActivity
import org.androidannotations.annotations.ViewById


@EActivity(R.layout.activity_search)
class SearchActivity : DrawerActivity() {

    @ViewById
    lateinit var svWord: SearchView
    @ViewById
    lateinit var wvDictOnline: WebView
    @ViewById
    lateinit var wvDictOffline: WebView

    @Bean
    lateinit var vm: SearchViewModel

    var word = ""
    var webViewFinished = false

    @AfterViews
    override fun afterViews() {
        super.afterViews()
        // http://stackoverflow.com/questions/3488664/android-launcher-label-vs-activity-title
        this.title = resources.getString(R.string.search)

        configWebView(wvDictOnline)
        configWebView(wvDictOffline)

        wvDictOnline.visibility = View.INVISIBLE
        wvDictOffline.visibility = View.INVISIBLE
        svWord.setQuery(vm.word, false)
        svWord.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String): Boolean {
                searchDict(svWord)
                return true
            }

            override fun onQueryTextChange(newText: String): Boolean {
                vm.word = newText
                return false
            }
        })
    }

    private fun configWebView(wv: WebView) {
        wv.settings.javaScriptEnabled = true // enable javascript
        val activity = this
        wv.webViewClient = object : WebViewClient() {
            override fun onReceivedError(view: WebView, errorCode: Int, description: String, failingUrl: String) {
                Toast.makeText(activity, description, Toast.LENGTH_SHORT).show()
            }
        }
    }

    fun searchDict(view: View?) {
        word = svWord.query.toString()
        wvDictOnline.visibility = View.VISIBLE
        wvDictOffline.visibility = View.INVISIBLE
        val url = vm.urlString!!
        svWord.post { svWord.clearFocus() }
        wvDictOnline.loadUrl(url)
    }

}
